javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/ConfiguredResourceAttributesHolder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.bootstrap;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.internal.ConfigPropertiesUtil;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public final class ConfiguredResourceAttributesHolder {
+
+  private static final Map<String, String> resourceAttributes = new HashMap<>();
+
+  public static Map<String, String> getResourceAttributes() {
+    return resourceAttributes;
+  }
+
+  public static void initialize(Attributes resourceAttribute) {
+    String[] mdcResourceAttributes = getConfiguredAttributes();
+
+    for (String key : mdcResourceAttributes) {
+      String value = resourceAttribute.get(stringKey(key));
+      if (value != null) {
+        ConfiguredResourceAttributesHolder.resourceAttributes.put(key, value);
+      }
+    }
+  }
+
+  private static String[] getConfiguredAttributes() {
+    String resourceAttributes =
+        ConfigPropertiesUtil.getString("otel.instrumentation.mdc.resource-attributes");
```

Comment:
I think it would be better to use `InstrumentationConfig.get().getList("otel.instrumentation.mdc.resource-attributes", emptyList())` but `InstrumentationConfig` isn't probably accessible from here. `InstrumentationConfig` gets the properties from the sdk and could support more source than what `ConfigPropertiesUtil` uses. `ConfigPropertiesUtil` is usually used for library instrumentations and `InstrumentationConfig` in javaagent. Could you try moving this class to https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/internal so you could use `InstrumentationConfig`. Sorry for not noticing this before, should be the last thing from me.